sopel/irc/__init__.py

```diff
@@ -301,6 +295,17 @@ def on_error(self):
         self.last_error_timestamp = datetime.now()
         self.error_count = self.error_count + 1
 
+    def change_current_nick(self, new_nick):
+        """Change the current nick without configuration modification.
+
+        :param str new_nick: New nick to be used by the bot.
+        """
+        self._nick = tools.Identifier(new_nick)
+        LOGGER.debug('Sending nick "%s"', self.nick)
+        self.backend.send_nick(self.nick)
+        LOGGER.debug('Sending user "%s" (name: "%s")', self.user, self.name)
+        self.backend.send_user(self.user, '+iw', self.nick, self.name)
```

Comment:
This is going to cause `ERR_ALREADYREGISTRED` when called later to regain the bot's configured nick after the ghost is killed or times out. No side effects, but it would be Better Behaved™ if Sopel sent this only when necessary.

Thanks to the handler for `RPL_WELCOME`, we already have a `bot.connection_registered` flag. But there's also no indication (to me) that a failed `NICK` command blocks the `USER` sent later in `on_connect()` here: https://github.com/sopel-irc/sopel/blob/f7bf3da022e75475c44ecbdad23edbbbe677a6fd/sopel/irc/__init__.py#L204-L207

That will still send the `USER` command even if `NICK` triggers `ERR_NICKNAMEINUSE` and fires off the handler in coretasks that comes back to call this function.

Based on all of that, I'm fairly confident that this suggestion will work just fine, but definitely test it if you apply it. 😝

```suggestion
```

Reply:
As discussed on IRC:

1. yes, it'll work
2. Modern IRC should be `USER <username> 0 * :<real name>` and we are sending `USER <username> +iw <nick> :<real name>` on connect so let's change that too.